src/main/java/org/broadinstitute/hellbender/tools/spark/transforms/bqsr/BaseRecalibratorEngineSpark.java

```diff
@@ -0,0 +1,125 @@
+import org.broadinstitute.hellbender.utils.recalibration.RecalibrationArgumentCollection;
+import org.broadinstitute.hellbender.utils.recalibration.RecalibrationTables;
+import org.broadinstitute.hellbender.utils.recalibration.covariates.StandardCovariateList;
+import org.broadinstitute.hellbender.utils.reference.ReferenceBases;
+import org.broadinstitute.hellbender.utils.variant.Variant;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Iterator;
+
+/**
+ * A lightweight wrapper over BaseRecalibrationEngine to make it easier to use from Spark.
+ * Takes in reads + contextual data (overlapping reference bases and variants), spits out RecalibrationTables.
+ */
+public final class BaseRecalibratorEngineSpark implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private Broadcast<SAMFileHeader> headerView;
+    private transient SAMFileHeader header;
+    private Broadcast<SAMSequenceDictionary> referenceSequenceDictionaryView;
+    private transient SAMSequenceDictionary referenceSequenceDictionary;
+
+    private RecalibrationArgumentCollection recalArgs;
+
+    private BaseRecalibrationEngine recalibrationEngine;
+
+    // true when we are about to process the first element of a bundle.
+    private transient boolean initialized = false;
+
+
+    // -------------------------------------------------------------
+    // public functions (and constructors, those go first)
+
+    /**
+     * DoFn for BaseRecalibrator.
+     * Takes in reads + contextual data (overlapping reference bases and variants), spits out RecalibrationTables.
+     */
+    public BaseRecalibratorEngineSpark(Broadcast<SAMFileHeader> headerView, Broadcast<SAMSequenceDictionary> referenceSequenceDictionaryView, RecalibrationArgumentCollection recalArgs) {
+        this.headerView = headerView;
+        this.referenceSequenceDictionaryView = referenceSequenceDictionaryView;
+        this.recalArgs = recalArgs;
+    }
+
+    // This ctor leaves referenceSequenceDictionaryView uninitialized. Caveat emptor.
+    private BaseRecalibratorEngineSpark(SAMFileHeader header, RecalibrationArgumentCollection recalArgs) {
+        this.header = header;
+        this.recalArgs = recalArgs;
+    }
+
+    // saves to output
+    public static void saveTextualReport(String output, SAMFileHeader header, RecalibrationTables rt, RecalibrationArgumentCollection recalArgs, AuthHolder auth) throws IOException {
+        OutputStream oStream = BucketUtils.createFile(output, auth);
+        QuantizationInfo qi = new QuantizationInfo(rt, recalArgs.QUANTIZING_LEVELS);
+        BaseRecalibratorEngineSpark worker = new BaseRecalibratorEngineSpark(header, recalArgs);
+        worker.onTraversalStart();
+        worker.saveReport(oStream, rt, qi, worker.recalibrationEngine.getCovariates());
```

Comment:
This also seems unnecessary -- you should be able to make `saveReport()` static, removing the need to create a new `BaseRecalibrationEngineSpark` instance here. See the final lines of `BaseRecalibratorSpark` for an example of how to avoid the `worker.recalibrationEngine.getCovariates()` call.